setup.py

```diff
@@ -41,7 +41,7 @@ def read(*filenames, **kwargs):
     "jupyter",
     "matplotlib>=2.0",
     "nbformat",
-    "pytest>=4.0,<5",
+    "pytest>=3.6",
```

Comment:
`pytest` 3.6 is the minimum required for Nengo. I tried the tests here with 3.6 and they worked, which is why I made the requirement 3.6. Do you know the reasoning behind the previous `>= 4.0` restriction?

I put no maximum version because we generally try to avoid that, as per the discussion in #264. I tried the newest `pytest`, and it works (that's also what the TravisCI tests will use, absent anything telling them not to). The reason we would have a maximum version is if a newer version of `pytest` was known to not work, not just because that *could* potentially happen at some point in the future.

Reply:
I guess, 4 was the lowest version I tried and I'm not too concerned to have the absolute minimum version for dev dependencies. If it works, it might be fine to bump the requirement down. But it also means that one has to be careful not to introduce any pytest 4 features (or remember to bump the version). Of course this is also a problem with version 4 and features from 5 etc. and the other dependencies. I wonder if there's a reasonable straight forward way to have a test run with the minium versions of the dependencies ...

Reply:
Yeah, it would be convenient to automate that. There's no obvious such function in `pip`, but it probably wouldn't be too hard to do something like this with `nengo-bones` (just by looking at the requirements defined in `.nengobones.yml` and pre-installing the minimum versions of any that have minimum versions specified).